database.py

```diff
@@ -21,27 +28,58 @@ def add_page(self, name):
             except Exception(exceptions.SessionExpired):
                 raise
 
-    def add_post(self, post_text):
+    def add_post(self, post_text, image):
         with self.driver.session() as session:
             try:
-                session.write_transaction(self.create_post_node, post_text)
+                session.write_transaction(self.create_post_node, post_text, image)
                 print(f"the post node has been created")
             except Exception(exceptions.SessionExpired):
                 raise
 
+    def add_relationship(self, page_name, post, time):
+        with self.driver.session() as session:
+            try:
+                session.write_transaction(self.create_relationship, page_name, post, time)
+                print(f"{page_name}--[{time}]-->{post}")
+            except Exception(exceptions.SessionExpired):
+                raise
+
     @staticmethod
     def create_page_node(tx, name):
         try:
-            return tx.run("CREATE (a:Page {name: $name})", name=name)
+            return tx.run("CREATE (a:Page {Name: $name})", name=name)
         except Exception(exceptions.CypherSyntaxError):
             raise
 
     @staticmethod
-    def create_post_node(tx, post_text):
+    def create_post_node(tx, post_text, image):
+        query = (
+            "CREATE (b:Post {Name: 'Post', Text: $text, Photo: $image})"
+            "RETURN b"
+        )
+        result = tx.run(query, text=post_text, image=image)
         try:
-            return tx.run("CREATE (b:Post {Name: 'Post', Text: $text})", text=post_text)
-        except Exception(exceptions.CypherSyntaxError):
+            return [{
+                "b": row["b"]["Post", "Text", "Photo"]
+            } for row in result]
+
+        except ServiceUnavailable as exception:
+            logging.error(f"{query} raised an error: \n {exception}")
             raise
 
+    @staticmethod
+    def create_relationship(tx, page_name, post, time):
+        query = (
+            "MATCH (a: Page), (b: Post)"
+            "WHERE a.Name = $name"
+            "CREATE (a)-[t:Date {Time:$time}]->(b)"
+            "RETURN t;"
+        )
+        result = tx.run(query, name=page_name, post=post, time=time)
+        try:
+            return result
+        except ServiceUnavailable as exception:
+            logging.error(f"{query} raised an error: \n {exception}")
+            raise
```

Comment:
made different methods to create relationships between page and its own posts, and it didn't work like how I wanted, so I have an idea to create relationships directly when the post node is created.